### PR TITLE
Added time from method overload for read by range 

### PR DESCRIPTION
**Pull request by patrickthoma**

### CODE REVIEW

File: BACnetClient.cs [hunk 1]
```diff
@@ -1197,11 +1208,38 @@ public void EndReadRangeRequest(IAsyncResult result, out byte[] trendbuffer, out
...
+                    getResult = () => (BacnetAsyncResult)BeginReadRangeRequest(adr, objectId, idxBegin, quantityCopy, true, invokeId);
+                    break;
+
+                default:
+                    throw new NotImplementedException($"BacnetReadRangeRequestTypes-Type {requestType} not supported in {nameof(ReadRangeRequestCore)}!");
+            }
+
             range = null;
-            using (var result = (BacnetAsyncResult)BeginReadRangeRequest(adr, objectId, idxBegin, quantity, true, invokeId))
+            using (var result = getResult())
```
Comment (gralin) [inline BACnetClient.cs:1242]: Just a suggestion to make the code shorter

```csharp
Func<IAsyncResult> getResult;
...
getResult = () => BeginReadRangeRequest(adr, objectId, readFrom, quantityCopy, true, invokeId);
getResult = () => BeginReadRangeRequest(adr, objectId, idxBegin, quantityCopy, true, invokeId);
...
using (var result = getResult() as BacnetAsyncResult)
```

File: BACnetClient.cs [hunk 2]
```diff
@@ -2579,5 +2617,6 @@ public void Dispose()
         {
             Transport.Dispose();
         }
+
```
Comment (gralin) [inline BACnetClient.cs:2620]: Please remove blank line

**Review by gralin — CHANGES_REQUESTED** [review]
Thank you @patrickthoma for your pull request! 🔥  Indeed the original implementation was limited (as it's the case no only here). I just suggested some cosmetic changed, hope you don't mind.

**Comment by patrickthoma** [comment]
Thank you very much for your input and your explanations, I changed the code accordingly.

**Review by gralin — APPROVED** [review]
Thanks!

**Pull request merged**